tools/rosmaster/src/rosmaster/registrations.py

```diff
@@ -464,7 +464,7 @@ def _register_node_api(self, caller_id, caller_api):
             else:
                 bumped_api = node_ref.api
                 self.thread_pool.queue_task(bumped_api, shutdown_node_task,
-                                            (bumped_api, caller_id, "new node registered with same name"))
+                                            (bumped_api, caller_id, "new node registered with same name {}".format(node_ref.id)))
```

Comment:
Instead of modifying the `reason` arg how about changing `shutdown()` call in the `shutdown_node_task()` function instead? In that function the `caller_id` could be prepended to the provided reason which makes sure that the node name is shown independent how the shutdown was triggered.

Reply:
Yeah I think that would solve the same problem in a better way. I'll make that change

Reply:
@cjds Friendly ping.

Reply:
Sorry I left this by the wayside. Made the change to `shutdown_node_task`. It now looks like

```
shutdown request: [/teleop_twist_keyboard] Reason: new node registered with same name
```